fix: WiFi/Bluetooth testing causes testService to hang

emilybrnkl opened the pull request:
The method of controlling serial ports caused the testService
to hang when data was not available on read.

HW-25

emilybrnkl committed:
fix: WiFi/Bluetooth testing causes testService to hang

The method of controlling serial ports caused the testService
to hang when data was not available on read.

HW-25

emilybrnkl committed:
Bump version to 3.0.2

dbeckwith01 commented on upload/upload.c:
```diff
@@ -33,8 +37,8 @@ void configureSerialPort(FlashState* s, int baud) {
   }
-  s->serialPort = open_serial(SERIAL_PORT_PATH, fd_convertBaud(baud));
-}
+  s->serialPort = fd_openSerial(SERIAL_PORT_PATH, MTK7697_BAUD);
```
It's better to use the baud rate passed into the function and not the define.

emilybrnkl replied:
Gotcha and fixed

dbeckwith01 commented on upload/upload.c:
```diff
@@ -67,34 +71,48 @@ void retryInitSequence(FlashState* s) {
     }
-    if (s->errorCount > 3 || (util_getUnixDatetime() - s->startTime > 3)) {
-      LE_INFO("Retrying...");
+    if((util_getUnixDatetime() - s->startTime > 3) && data == 0){
```
What if I get another character besides 'C' or 0?  Shouldn't we just give up after 3 attempts?

emilybrnkl replied:
Edited to !='0' as an error since C is the only valid case

dbeckwith01 reviewed (changes requested):
A minor change and something to think about.

emilybrnkl committed:
fix: WiFi/Bluetooth testing causes testService to hang

Editing test case to exit after three failed attempts
at serial communication

HW-25
